solarkraft/src/fetcher/callDecoder.ts

```diff
@@ -23,17 +23,24 @@ export async function extractContractCall(
     op: any,
     matcher: (contractId: string) => boolean
 ): Promise<Maybe<ContractCallEntry>> {
-    // https://developers.stellar.org/network/horizon/api-reference/resources/operations/object
+    // https://developers.stellar.org/network/horizon/api-reference/resources/operations/object/invoke-host-function
     if (op.function !== 'HostFunctionTypeHostFunctionTypeInvokeContract') {
         return none()
     }
 
     // This operation represents a smart contract call.
     // In particular, it contains: the callee contract, the method, and its parameters.
     //
+    // https://developers.stellar.org/network/horizon/api-reference/resources/operations/object
     // https://developers.stellar.org/network/horizon/api-reference/resources/operations/object/invoke-host-function
 
-    // extract the call data: transaction hash, parameters
+    // Extract the call data: time, transaction hash, parameters
+
+    // In Soroban, `env.ledger().timestamp()` "[r]eturns a unix timestamp for when the ledger was closed":
+    // https://docs.rs/soroban-sdk/latest/soroban_sdk/ledger/struct.Ledger.html#method.timestamp
+    // according to [this](https://stellar.stackexchange.com/questions/1852/transaction-created-at-and-ledger-close-time),
+    // the transaction object's `created_at` equals the ledger's `closed_at`.
+    const timestamp = new Date(op.created_at).getTime() / 1000
```

Comment:
Do we need the `/1000` here? I understand that it says [here](https://docs.rs/soroban-sdk/latest/soroban_sdk/ledger/struct.Ledger.html#method.timestamp) that the time is given in seconds, but in a degenerate case it could happen that `Date(t1).getTime()` and `Date(t2).getTime()` are different in `ms` units but same in `s` units due to truncation.

Reply:
Horizon only gives us seconds precision, it's just that `Date` (which we use for parsing) internally represents milliseconds.

We could leave it as milliseconds here, but I think it's preferable to have parity between the spec and the contract code (in case there are any constants).

I agree though that this looks confusing. I'll add a comment for clarification.